predicators/pretrained_model_interface.py

```diff
@@ -249,3 +253,135 @@ def _sample_completions(
+                "type": "image_url"
+            })
+
+        if suffix:
+            content.append({"text": suffix, "type": "text"})
+
+        return [{"role": "user", "content": content}]
+
+    @retry(wait=wait_random_exponential(min=1, max=60),
+           stop=stop_after_attempt(6))
+    def call_openai_api(self,
+                        messages: list,
+                        model: str = "gpt-4",
+                        seed: Optional[int] = None,
+                        max_tokens: int = 32,
+                        temperature: float = 0.2,
+                        verbose: bool = False):
+        """Make an API call to OpenAI."""
+        client = openai.OpenAI()
+        completion = client.chat.completions.create(
+            model=model,
+            messages=messages,
+            seed=seed,
+            max_tokens=max_tokens,
+            temperature=temperature,
+        )
+        if verbose:
+            print(f"OpenAI API response: {completion}")
+        assert len(completion.choices) == 1
+        return completion.choices[0].message.content
+
+    def get_id(self) -> str:
+        """Get an identifier for the model."""
+        return f"OpenAI-{self.model_name}"
+
+    def _sample_completions(
+        self,
+        prompt: str,
+        imgs: Optional[List[PIL.Image.Image]],
+        temperature: float,
+        seed: int,
+        stop_token: Optional[str] = None,
+        num_completions: int = 1,
+        max_tokens=512,
+    ) -> List[str]:
+        """Query the model and get responses."""
+        messages = self.prepare_vision_messages(prefix=prompt,
+                                                images=imgs,
+                                                detail="auto")
+        responses = [
+            self.call_openai_api(messages,
+                                 model=self.model_name,
+                                 max_tokens=max_tokens,
+                                 temperature=temperature)
+            for _ in range(num_completions)
+        ]
+        return responses
+
+
+# Example usage:
+if __name__ == "__main__":
+    # Make sure the OPENAI_API_KEY is set
+    model_name = "gpt-4-turbo"
+    vlm = OpenAIVLM(model_name)
+
+    prompt = """
+        Describe the object relationships between the objects and containers.
+        You can use following predicate-style descriptions:
+        Inside(object1, container)
+        Blocking(object1, object2)
+        On(object, surface)
+        """
+    images = [PIL.Image.open("../tests/datasets/test_vlm_predicate_img.jpg")]
+
+    print("Start requesting...")
+    completions = vlm.sample_completions(prompt=prompt,
+                                         imgs=images,
+                                         temperature=0.5,
+                                         num_completions=3,
+                                         seed=0)
+    for i, completion in enumerate(completions):
+        print(f"Completion {i + 1}: \n{completion}\n")
```

Comment:
remove; we can show usage somewhere else/in the test

Reply:
sounds good; I can move it to test? maybe a separate test function but without actually requesting?